### ISSUE TITLE
Quirk on retrieving via json pointer syntax

### ISSUE THREAD

**Issue by stephenberry** [stephenberry/glaze#1217]
### Discussed in https://github.com/stephenberry/glaze/discussions/1195

<div type='discussions-op-text'>

<sup>Originally posted by **raddy** July 21, 2024</sup>
The behavior of get_sv_json behaves a bit differently in some cases when retrieving the last key of a JSON blob -- the newline and spaces are attached. 

If the object is inside an array like:

`std::string json = R"({
        "data": [
            {
                "id": 162904329731651
            },
            {
                "id": 162904329751651
            },
        ]
    })";`

We parse the `id` as `id: 162904329751651N____________` (replacing \n with N and ' ' with '_' for visibility)

A minimal example: [godbolt](https://gcc.godbolt.org/z/b96W5bqeY)

Relatedly, is there a preferred way to use get_sv_json to loop through such an array? I've been doing manual string splitting (as I expect using the array notation takes me back to the beginning each time?)</div>

**Comment by stephenberry** [comment]
I finally got a chance to look at this bug and it was because we had never used `get_sv_json` with formatted JSON, which it ought to work with.

I merged a fix (#1271), which will better validate the value you read with `get_sv_json` and eliminate the trailing whitespace that you were seeing.

As to your other question:
> Relatedly, is there a preferred way to use get_sv_json to loop through such an array? I've been doing manual string splitting (as I expect using the array notation takes me back to the beginning each time?)

`get_sv_json` is going to start parsing from the top of the path each time. What I would do is use the path to the array to get the entire array as a string_view (`"/data"`) and then use partial reading with Glaze to read in the contents you want. Albeit this only would work for reading some number of initial values and wouldn't let you skip to some arbitrary point in the array.
